pkg/agent/route/route_linux.go

```diff
@@ -1434,21 +1421,30 @@ func (c *Client) AddLoadBalancer(svcIP net.IP) error {
 // gateway on host.
 func (c *Client) DeleteLoadBalancer(svcIP net.IP) error {
 	svcIPStr := svcIP.String()
-	route, found := c.serviceRoutes.Load(svcIPStr)
-	if !found {
-		klog.V(2).InfoS("Didn't find route for LoadBalancer ingress IP", "ingressIP", svcIPStr)
-		return nil
+	if err := c.deleteServiceRoute(svcIP); err != nil {
+		return fmt.Errorf("failed to delete route for LoadBalancer ingress IP %s: %w", svcIPStr, err)
 	}
-	if err := c.netlink.RouteDel(route.(*netlink.Route)); err != nil {
-		if err.Error() == "no such process" {
-			klog.InfoS("Failed to delete route for LoadBalancer ingress IP since it doesn't exist", "route", route)
-		} else {
-			return fmt.Errorf("failed to delete route for LoadBalancer ingress IP %s: %w", svcIPStr, err)
-		}
+	klog.V(4).InfoS("Deleted route for LoadBalancer ingress IP", "IP", svcIPStr)
+	return nil
+}
+
+// AddExternalIP is used to add routing entry which is used to route ExternalIP to Antrea gateway on host.
+func (c *Client) AddExternalIP(svcIP net.IP) error {
```

Comment:
How about naming it AddServiceExternalIPRoute or AddExternalIPService to be clearer?

Reply:
Good idea.